### PR TITLE
Fix @ForeignKeyey(ConstraintMode.NO_CONSTRAINT) throwing EntityNotFou…

### PR DESCRIPTION
**Pull request by dreab8**
…ndException

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java [hunk 1]
```diff
@@ -132,6 +132,7 @@
 	 */
 	private final boolean isKeyTableNullable;
 	private final boolean isConstrained;
+	private final boolean isForeignKeyEnabed;
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:135]: Actually, `isConstrained` should be enough. I don't think you need another field, or am I missing something?

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:135]: unfortunately for a `OneToOne` is not, I tried to use it but a I had a lot of failures :(

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:135]: @beikov by the way it seems this solution is not working on top of the latest changes... I'm working on a slightly different solution

**Review by beikov — APPROVED** [review]
LGTM

**Comment by sebersole** [comment]
Is this still necessary given the work I did on `@NotFound`?  That work also updated how `ConstraintMode.NO_CONSTRAINT` is handled in certain respects.

**Comment by dreab8** [comment]
I reverted my changes and run the tests, the 2 tests I removed the `@NotImplementedYet` still fail.

**Comment by beikov** [comment]
The fix looks good to me and the description of what happens make sense. Do you still have concerns @sebersole?

**Review by sebersole — APPROVED** [review]
LGTM

**Pull request merged**